src/Kestrel.Core/KestrelServer.cs

```diff
@@ -52,6 +54,8 @@ internal KestrelServer(ITransportFactory transportFactory, ServiceContext servic
             Features = new FeatureCollection();
             _serverAddresses = new ServerAddressesFeature();
             Features.Set(_serverAddresses);
+
+            _defaultHttpsProvider = serviceContext.ServerOptions.ApplicationServices.GetService<IDefaultHttpsProvider>();
```

Comment:
Don't call GetService for this. Just make IDefaultHttpsProvider a param to the public ctor.